src/clib/pioc.c

```diff
@@ -1808,7 +1808,7 @@ int PIOc_init_async(MPI_Comm world, int num_io_procs, int *io_proc_list,
         }
     }
     else
-        my_proc_list = proc_list;
+        my_proc_list = (int**)proc_list;
```

Comment:
Are we casting away the constness here?

Reply:
There is a compiler warning here without this explicit casting. The compiler always drops the constness when assigning proc_list to my_proc_list (implicit casting).

Reply:
Maybe we can define my_proc_list as const int** instead.

Reply:
Another way to tackle this would be to have my_proc_list as const int **. You would need to point it to a temp malloc'ed buffer (in case the user does not provide the proc list) if needed.

Reply:
OK, I will update it later (always try to preserve constness).

Reply:
I have updated this PR to have my_proc_list as const int **.